File: two-pointers/283. Move Zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function (nums) {
  let left = 0, right = 1;
  
  while (right < nums.length) {
    if (nums[left] === 0) {
      if (nums[right] !== 0) {
        nums[left] = nums[right];
        nums[right] = 0;
        left++;
        right++;
        continue;
      }
      right++;
      continue
    }
    right++;
    left++;
  }
};

// Test
// const nums = [0,1,0,3,12];
const nums = [0];
moveZeroes(nums);
// tslint:disable-next-line:no-console
console.log('moveZeroes: ', nums); // eslint-disable-line no-console

File: python/two_pointers/283. Move Zeroes.py
from typing import List


class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        # new = [x for x in nums if x != 0]
        # for i in range(0, len(nums)):
        #     if i < len(new):
        #         nums[i] = new[i]
        #     else:
        #         nums[i] = 0

        left = 0
        for right in range(len(nums)):
            if nums[right] != 0:
                nums[left], nums[right] = nums[right], nums[left]
                left += 1

